modules/internal/DefaultRectangular.chpl

```diff
@@ -114,10 +114,6 @@ module DefaultRectangular {
     proc dsiGetIndices() return ranges;
   
     proc dsiSetIndices(x) {
```

Comment:
I am on the fence about removing the checks below. My inclination to keep them, and add one for stridability, is because dsiSetIndices gets invoked, perhaps via setIndices(), in a couple of places where conformance is not guaranteed, it seems. It would be nice to get clear error diagnostics in such cases. Delegating that error checking to range assignments would result in a somewhat obscure error message - range assignment will be reported, whereas the user is doing domain assignment. IMO it is worth having duplicated checking for the above benefits.